src/OpenTelemetry.Instrumentation.Http/CHANGELOG.md

```diff
@@ -2,6 +2,13 @@
 
 ## Unreleased
 
+* Spans will now be exported even if the request was instrumented before. This
+is to enable exporting spans in case of retries as per the
+[spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-retries-examples)
+where same request can be reused. Note: This change will not create a new span
```

Comment:
The spec states each retry should have a new activity. looking at https://github.com/open-telemetry/opentelemetry-dotnet/pull/2756, I thought the plan was to create these activities inside instrumentation library. 

Just thinking about it again, the change in this PR does not necessarily relates to retry. It would also cover the case where the request was already instrumented before by some other library than OTel. In this case we were not exporting the span, with this change we will.